### PR TITLE
Adds redis for session storage.

### PR DESCRIPTION
**Pull request by cmacfarl**
Will default to a local sqlite database if the redis ip address is not set in the environment.

Tested locally on Windows using an Ubuntu WSL instance with redis-server installed.

### CODE REVIEW

File: server/credentialstore.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import redis
+import constants
+
+class CredentialStore():
+
+    def __init__(self):
+        self.red = redis.Redis(host=constants.REDIS_IP_ADDR, port=6379)
```
Comment (lizlooney) [inline server/credentialstore.py:21]: Is the port always 6379? Even on production servers?

File: terraform/core/main.tf [hunk 2]
```diff
@@ -188,6 +189,7 @@ resource "google_app_engine_standard_app_version" "fmltc-app-v1" {
   env_variables = {
     PROJECT_ID = var.project_id
     ORIGIN = "https://${var.project_id}.uc.r.appspot.com"
+    USE_OIDC = "true"
```
Comment (lizlooney) [inline terraform/core/main.tf:192]: Do you need to set REDIS_IP_ADDR here? Is REDIS_IP_ADDR only for local use?

File: server/credentialstore.py (hunk 1 above)
Comment (cmacfarl) [reply server/credentialstore.py:21]: 6379 is the default port for redis servers.  And I've never seen a redis server running on anything other than 6379, including the redis servers used for ftc-scoring and all the local development servers I've run.  I think the only reason to specifically define a port other than 6379 is if you want to run two redis servers on the same machine.  I think this is fine for now.  I can add it to constants if you really want, but that's one more thing to break if not set properly.

File: terraform/core/main.tf (hunk 2 above)
Comment (cmacfarl) [reply terraform/core/main.tf:195]: Yes.  That's in a820c6809cf8893ee579a6f02f7270c9ee73045c

**Pull request merged**